src/Features/Core/Portable/SpellCheck/AbstractSpellCheckCodeFixProvider.cs

```diff
@@ -0,0 +1,118 @@
+
+            var syntaxRoot = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var node = syntaxRoot.FindNode(span);
+            if (node == null || node.Span != span)
+            {
+                return;
+            }
+
+            SemanticModel semanticModel = null;
+            foreach(var name in node.DescendantNodesAndSelf().OfType<TSimpleName>())
+            {
+                // Only bother with identifiers that are at least 3 characters long.
+                // We don't want to be too noisy as you're just starting to type something.
+                var nameText = name.GetFirstToken().ValueText;
+                if (nameText.Length >= 3)
+                {
+                    semanticModel = semanticModel ?? await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+                    var symbolInfo = semanticModel.GetSymbolInfo(name, cancellationToken);
+                    if (symbolInfo.Symbol == null)
+                    {
+                        await CreateSpellCheckCodeIssueAsync(context, name, nameText, cancellationToken).ConfigureAwait(false);
+                    }
+                }
+            }
+        }
+
+        private async Task CreateSpellCheckCodeIssueAsync(CodeFixContext context, TSimpleName nameNode, string nameText, CancellationToken cancellationToken)
+        {
+            var document = context.Document;
+            var completionList = await CompletionService.GetCompletionListAsync(
+                document, nameNode.SpanStart, CompletionTriggerInfo.CreateInvokeCompletionTriggerInfo(), cancellationToken: cancellationToken).ConfigureAwait(false);
+            if (completionList == null)
+            {
+                return;
+            }
+
+            var completionRules = CompletionService.GetCompletionRules(document);
+            var onlyConsiderGenerics = IsGeneric(nameNode);
+            var results = new MultiDictionary<double, string>();
+
+            int closeMatchThreshold = EditDistance.GetCloseMatchThreshold(nameText);
+
+            foreach (var item in completionList.Items)
+            {
+                if (onlyConsiderGenerics && !IsGeneric(item))
+                {
+                    continue;
+                }
+
+                var candidateText = item.FilterText;
+                double matchCost;
+                if (!EditDistance.IsCloseMatch(nameText, candidateText, closeMatchThreshold, out matchCost))
+                {
+                    continue;
+                }
+
+                var insertionText = completionRules.GetTextChange(item).NewText;
+                results.Add(matchCost, insertionText);
+            }
+
+            var matches = results.OrderBy(kvp => kvp.Key)
+                                 .SelectMany(kvp => kvp.Value)
+                                 .Where(t => t != nameText)
+                                 .Take(3)
```

Comment:
Sure.  I can do that.  I'm not generally a fan of constants for things used only once.  But I see no harm here in making that change.